ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,170 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let create =
+  call ~name:"create" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (String, "name_label", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ; (Bool, "status", "The status of the TracerProvider")
+      ; (Map (String, String), "tags", "The tags of the TracerProvider")
+      ; (Set String, "endpoints", "The endpoints of the TracerProvider")
+      ; (Set String, "components", "The componenets of the TracerProvider")
+      ; (Set String, "filters", "The filters of the TracerProvider")
+      ; (Set String, "processors", "The processors of the TracerProvider")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let destroy =
+  call ~name:"destroy" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:[(Ref _tracing, "self", "The TracerProvider to be destoryed")]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_hosts =
+  call ~name:"set_hosts" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the host field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_status =
+  call ~name:"set_status" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the status of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; (Bool, "status", "The status the TracerProvider is to be set to")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_tags =
+  call ~name:"set_tags" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the tags field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Map (String, String)
+        , "tags"
+        , "The tags the TracerProvider is to be set to"
```

Comment:
The documentation should try to explain the purpose of tags as otherwise it does not add much beyond what the function name already suggests. "Set the tags emitted by this provider" - or similar.

Reply:
Resolved by: [0d0b6af](https://github.com/xapi-project/xen-api/pull/4966/commits/0d0b6aff8e9aaf3ff50907258e3ddd71ad5c0978)